library/src/scala/annotation/RefiningAnnotation.scala

```diff
@@ -0,0 +1,7 @@
+package scala.annotation
+
+/** A base trait for annotations that yield proper subtypes of the types they annotate.
+ *  Refining annotations are more "sticky" than normal ones. They are conceptually kept
+ *  around when normal refinements would also not be stripped away.
+ */
+trait RefiningAnnotation extends StaticAnnotation
```

Comment:
Why expose that in the stdlib, isn't this compiler internal only?

Reply:
No, you might want to define your own annotations (say units of measure). Maybe they need a compiler plugin to  order to be checked, or maybe you can do it in the future with annotation macros. `RefiningAnnotation` just makes sure they propagate correctly.